client/me/purchases/components/payment-method-form/index.jsx

```diff
@@ -257,18 +258,23 @@ function StripeError( { translate } ) {
 	);
 }
 
-function displayError( { translate, error } ) {
+function displayError( { translate, error }, reduxDispatch ) {
 	if ( error instanceof StripeSetupIntentError || error instanceof StripeValidationError ) {
-		notices.error( <StripeError translate={ translate } /> );
+		reduxDispatch( errorNotice( <StripeError translate={ translate } /> ) );
 		return;
 	}
 	if ( typeof error.message === 'object' ) {
-		notices.error( <ValidationErrorList messages={ values( error.message ) } /> );
+		reduxDispatch( errorNotice( <ValidationErrorList messages={ values( error.message ) } /> ) );
 		return;
 	}
-	notices.error( error.message );
+	reduxDispatch( errorNotice( error.message ) );
 }
 
-export default connect( ( state ) => ( {
-	countriesList: getCountries( state, 'payments' ),
-} ) )( localize( PaymentMethodForm ) );
+export default connect(
+	( state ) => ( {
+		countriesList: getCountries( state, 'payments' ),
+	} ),
+	{
+		errorNotice,
```

Comment:
This component is a little unusual in that it's a functional component using `connect` rather than `useSelector` to connect to Redux. In this PR, we add `useDispatch` and pass it the raw action creator, but here we bind the action creator to a custom dispatcher in the component's props, which (as far as I can tell) is not used.

I think we should commit to either using hooks or `connect` for this component to avoid confusion in the future. I'm fine with either but I do tend to prefer hooks in general; if you agree, we could remove `connect` here entirely and add `const countriesList = useSelector( ( state ) => getCountries( state, 'payments' ) );` to the top of the component. 

Alternatively, you could remove the call to `useDispatch` and instead pass the `errorNotice` prop to `displayError`. This would be a little difficult to avoid a shadow variable since the prop has the same name as the import, so you'd probably need to alias it either here or when the props are destructured.

(The third option is just to remove the `mapDispatchToProps` argument here and leave `useDispatch` above, but I personally think it's just going to cause more confusion later if we don't pick one pattern now.)

Reply:
🤦  good catch!

I initially did use `connect` because that's my gut intuition when I see it used. Then I used the hook and forgot to remove the bound action from `connect`. 

I agree it's best to commit to hooks in this case, so I've removed the `mapDispatchToProps` I introduced in 54871ea33d8433432b36f9203b96b631531c0672. Furthermore, to remove the pre-existing confusion, I've removed the `connect` usage and replaced it with a `useSelector` use as you suggested in 5b0cf15fc04c02f742bb5fef426b12bcaeaa27cc